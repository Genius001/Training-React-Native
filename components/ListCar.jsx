import { Text, StyleSheet, Image } from 'react-native';
import React from 'react';
import { Row, Col } from '@/components/Grid';
import { Feather } from '@expo/vector-icons';
import Button from './Button';

const formatCurrency = new Intl.NumberFormat('id-ID', {
  style: 'currency',
  currency: 'IDR',
});

export default function ListCar({
  onPress,
  image,
  carName,
  passengers,
  baggage,
  price,
  style,
}) {
  const formatIDR = (price) => formatCurrency.format(price);

  return (
    <Button
      style={{ ...styles.card, ...style }}
      onPress={onPress}>
      <Row alignItems={'center'} gap={20}>
        <Col>
          <Image style={styles.img} source={image} resizeMode="contain" />
        </Col>
        <Col>
          <Text style={styles.carName}>{carName}</Text>
          <Row>
            <Col style={styles.textIcon}>
              <Feather name="users" size={12} color="#8A8A8A" />
              <Text style={styles.capacityText}>{passengers}</Text>
            </Col>
            <Col style={styles.textIcon}>
              <Feather name="briefcase" size={12} color="#8A8A8A" />
              <Text style={styles.capacityText}>{baggage}</Text>
            </Col>
          </Row>
          <Text style={styles.price}>{formatIDR(price)}</Text>
        </Col>
      </Row>
    </Button>
  );
}

const styles = StyleSheet.create({
  card: {
    backgroundColor: '#FFFFFF',
    borderRadius: 4,
    shadowColor: '#000000',
    shadowOpacity: 0.15,
    elevation: 20,
    padding: 20,
    marginBottom: 20,
    width: '100%',
    alignItems: 'left',
  },
  img: {
    width: 60,
    height: 60,
  },
  carName: {
    fontSize: 14,
    fontFamily: 'Poppins',
    fontWeight: '400',
    color: '#000000',
  },
  capacityText: {
    color: '#8A8A8A',
    fontFamily: 'Poppins',
    fontSize: 12,
    fontWeight: '700',
    marginLeft: 4,
  },
  price: {
    color: '#5CB85F',
    fontFamily: 'Poppins',
    fontSize: 14,
  },
  textIcon: {
    flexDirection: 'row',
    alignItems: 'center',
    gap: 2,
  },
});
