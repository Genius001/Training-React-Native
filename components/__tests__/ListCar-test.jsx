import React from 'react';
import renderer from 'react-test-renderer';
import ListCar from '../ListCar';

it(`renders correctly`, () => {
  const tree = renderer.create(
    <ListCar
      image={require("@/assets/images/img_photo.png")}
      carName={"Innova"}
      passengers={5}
      baggage={2}
      price={500000}
    />
  ).toJSON();


  expect(tree).toMatchSnapshot();
});
