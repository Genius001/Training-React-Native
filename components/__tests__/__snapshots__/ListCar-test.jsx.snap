// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  jestAnimatedStyle={
    {
      "value": {},
    }
  }
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "left",
      "backgroundColor": "#FFFFFF",
      "borderRadius": 4,
      "color": "#fff",
      "elevation": 20,
      "marginBottom": 20,
      "opacity": 1,
      "padding": 20,
      "paddingVertical": 8,
      "shadowColor": "#000000",
      "shadowOpacity": 0.15,
      "width": "100%",
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "gap": 20,
        "justifyContent": "flex-start",
      }
    }
  >
    <View>
      <Image
        resizeMode="contain"
        source={1}
        style={
          {
            "height": 60,
            "width": 60,
          }
        }
      />
    </View>
    <View>
      <Text
        style={
          {
            "color": "#000000",
            "fontFamily": "Poppins",
            "fontSize": 14,
            "fontWeight": "400",
          }
        }
      >
        Innova
      </Text>
      <View
        style={
          {
            "alignItems": "baseline",
            "flexDirection": "row",
            "gap": 0,
            "justifyContent": "flex-start",
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "flexDirection": "row",
              "gap": 2,
            }
          }
        >
          <Text />
          <Text
            style={
              {
                "color": "#8A8A8A",
                "fontFamily": "Poppins",
                "fontSize": 12,
                "fontWeight": "700",
                "marginLeft": 4,
              }
            }
          >
            5
          </Text>
        </View>
        <View
          style={
            {
              "alignItems": "center",
              "flexDirection": "row",
              "gap": 2,
            }
          }
        >
          <Text />
          <Text
            style={
              {
                "color": "#8A8A8A",
                "fontFamily": "Poppins",
                "fontSize": 12,
                "fontWeight": "700",
                "marginLeft": 4,
              }
            }
          >
            2
          </Text>
        </View>
      </View>
      <Text
        style={
          {
            "color": "#5CB85F",
            "fontFamily": "Poppins",
            "fontSize": 14,
          }
        }
      >
        RpÂ 500.000,00
      </Text>
    </View>
  </View>
</View>
`;
